### ISSUE TITLE
Public Client Property while creating AD application (azuread_application)

### ISSUE THREAD

**Issue by saiprasanthrajavarapu** [terraform-providers/terraform-provider-azuread#74]
Issues on GitHub are intended to be related to bugs or feature requests with provider codebase,
so we recommend using our other community resources instead of asking here 👍.

Hello Team,

we are using "azuread_application" provider to create a client Azure AD application. we dont see any Public Client property in attributes of the provider. Is there a way to achieve a public client property using terraform? 

```
resource "azuread_application" "clientreg" {
  name                       = "${local.Clientname}"
  reply_urls                 = ["https://${local.Clientname}"]
  available_to_other_tenants = true
  required_resource_access {
    resource_app_id = "00000000-0000-0000-0000-000000000000"
  
    resource_access {
      id = "${local.AzureAD}"
      type = "Scope"
    }
  }
  required_resource_access {
    resource_app_id = "${azuread_application.serverreg.application_id}"
  
    resource_access {
      id = "${data.external.AADserver_application.result["oauth2PermissionsID"]}"
      type = "Scope"
    }
  }
  lifecycle {
    ignore_changes = ["required_resource_access","tags.CreatedDate"]
  }
}
```

**Comment by lzadjsf** [comment]
Seems like duplicate...
https://github.com/terraform-providers/terraform-provider-azuread/issues/67
Please vote there

**Comment by athempel** [comment]
No, this is not a duplicate of #67.

#67 is about retrieving the application ID of an Azure AD application.

This is about setting the "allowPublicClient" boolean property of an Azure AD application.

**Comment by katbyte** [comment]
Hi @athempel,

This will be exposed by setting the "type" of the application. This matches the Azure portal and currently being added via #13.

**Comment by katbyte** [comment]
Closed by #34
